pandas/core/indexes/base.py

```diff
@@ -5589,7 +5588,8 @@ def set_value(self, arr, key, value):
             stacklevel=find_stack_level(),
         )
         loc = self._engine.get_loc(key)
-        validate_numeric_casting(arr.dtype, value)
+        if not can_hold_element(arr, value):
+            raise ValueError
```

Comment:
caught elsewhere?

Reply:
set_value is deprecated and we don't use it, this is just retaining old behavior.  (i guess we could keep validate_numeric_casting here until the deprecation is enforced (im trying to get rid of it entirely))